wallet/how-to/use-non-evm-networks/starknet/manage-starknet-networks.md

```diff
@@ -0,0 +1,44 @@
+---
+description: Manage Starknet networks in MetaMask.
+sidebar_position: 3
+---
+
+# Manage Starknet networks
+ 
+## Switch between networks
+
+Allow users to switch between different Starknet networks, such as mainnet and testnet. This function enables switching by setting the appropriate network RPC URL.
+
+```javascript
+const switchNetwork = async (network) => {
+  try {
+    const starknet = getStarknet();
+
+    // Example network URLs
+    const networks = {
+      mainnet: 'https://mainnet.starknet.io',
+      testnet: 'https://sepolia.starknet.io'
+    };
+
+    // Set the provider based on the selected network
+    starknet.setProvider({ rpcUrl: networks[network] });
+
+    console.log(`Switched to ${network}:`, starknet.provider);
+  } catch (error) {
+    console.error('Error switching network:', error);
+  }
+};
```

Comment:
In case we are in the same set up than the tutorial at the beginning and we have access to the wallet object we stored in state. Switching chain is as simple as : 

```suggestion
const switchChain = async (chainId: string) => {
    try {
        await wallet?.request({
            type: 'wallet_switchStarknetChain',
            params: { chainId: chainId },
        });
        console.log(`Switched to chainId: ${chainId}`);
    } catch (e) {
        console.error("Failed to switch chain:", e);
    }
};
```